spec/httpserver/handlers/handler_spec.clj

```diff
@@ -1,13 +1,27 @@
 (ns httpserver.handler_spec
    (:require [speclj.core :refer :all]
-   [httpserver.handlers.handler :as handler]))
+   [httpserver.handlers.handler :as handler]
+   [httpserver.handlers.teapot :as teapot]
+   [httpserver.routing.router :as router]
+   ))
 
 (import '(com.td.HttpServer HttpResponse))
 (import '(com.td.HttpServer HttpRequest))
 
-(describe "main handler"
+(describe "generating a response"
   (it "returns a HttpResponse with code 418 when passed a HttpRequest GET /coffee"
-    (let [httpRequest (HttpRequest. "GET /coffee HTTP/1.1" (java.util.HashMap. {"this" "that"}))
-          main-handler (handler/new-handler)
-          httpResponse (.generateResponse main-handler httpRequest)]
+      (let [test-list (list {:method "GET" :path "/coffee" :handler teapot/call}
+                            {:method "GET" :path "/test"}
+                            {:method "PUT" :path "/test"})
```

Comment:
These two routes are unnecessary for the test. I think you can just remove them and just have `test-list` be a one-element vector